packages/jest-haste-map/src/ModuleMap.ts

```diff
@@ -3,52 +3,49 @@
  *
  * This source code is licensed under the MIT license found in the
  * LICENSE file in the root directory of this source tree.
- *
- * @flow
  */
 
-import type {Path} from 'types/Config';
-import type {
+import {Config} from '@jest/types';
+import {
   DuplicatesSet,
   HTypeValue,
   ModuleMetaData,
   RawModuleMap,
   ModuleMapData,
   DuplicatesIndex,
   MockData,
-} from 'types/HasteMap';
+} from './types';
 
 import * as fastPath from './lib/fast_path';
 import H from './constants';
 
-const EMPTY_OBJ = {};
+const EMPTY_OBJ = {} as {[key: string]: any};
 const EMPTY_MAP = new Map();
 
-export opaque type SerializableModuleMap = {
+type ValueType<T> = T extends Map<T, infer V> ? V : T;
+
+export type SerializableModuleMap = {
   // There is no easier way to extract the type of the entries of a Map
```

Comment:
I get type errors going back and forth, so even though I've simplified the type here, it's wrong. which is why I kept the comment 😅